pkg/cache/unavailableofferings.go

```diff
@@ -31,27 +32,43 @@ import (
 type UnavailableOfferings struct {
 	// key: <capacityType>:<instanceType>:<zone>, value: struct{}{}
 	cache *cache.Cache
+	// pre-computed key for spot since we will be computing this key many times and it
+	// will always be the same.
+	spotKey string
 }
 
 func NewUnavailableOfferingsWithCache(c *cache.Cache) *UnavailableOfferings {
-	return &UnavailableOfferings{
-		cache: c,
+	u := &UnavailableOfferings{
+		cache: cache.New(UnavailableOfferingsTTL, DefaultCleanupInterval),
 	}
+	u.spotKey = u.key("", "", v1beta1.CapacityTypeSpot)
```

Comment:
If you change `key()` to non-member (which should not be a problem), all this unpleasantness would go away